### ISSUE TITLE
Interactive Data Table

### ISSUE THREAD

**Issue by ngodber** [keplergl/kepler.gl#1573]
**Is your feature request related to a problem? Please describe.**
A common paradigm in GIS solutions is to allow the user to select features on a layer and view the properties in a table. That table generally allows the user to order by different columns and supports export  those points to csv or wkt etc
![image](https://user-images.githubusercontent.com/11834846/129278494-6aec66b8-7179-402f-9900-0e5c8c42f6dd.png)

The frustration is that when a User is visualizing their data once they've gained some insight and found an area of interest they aren't able to translate that learning into further action. You cannot further interact with the data in ways that allow you to capture that learning - select and export features of interest.

**Describe the solution you'd like**
The has the ability to select using polygon or square as is currently the case and include option to pop up table showing the properties of the selected points. Table would also have functionality to order by columns and to export rows to text.

**Describe alternatives you've considered**
The closest functionality is the tool tip. However this is feature by feature and still gives not ability to actually save this data out.

**Comment by ngodber** [comment]
@heshan0131 is there a major roadblock to this functionality that I am unaware of?

**Comment by heshan0131** [comment]
> Hi Shan,
> 
> Thanks for replying! First of all, I just want to express how much of a massive fan I am of everything you guys have done first at Uber and now at Unfolded/foursquare. Some of your data visualizations border on art.
> 
> I'm happy to commit some time to implement 2. Whilst on the topic would there be interest in replacing the table with something more features. Googling indicates that https://github.com/Autodesk/react-base-table has good support for large datasets and includes some nice features out of the box. Its also MIT so compatible with Kepler license.

Sure that table component looks great. What I am interested is the ability to Select rows in side the table component. The current table component doesn't support that.

To add a toggle in the data table to should filtered data. you will first need to call `applyCPUFilter` action (see how it is used in `ExportDataModal`), when called, dataset will be filtered, and you can get the filtered data by accessing `dataset.filteredIdxCPU`, it's an array of indexes of filtered data.

**Comment by ngodber** [comment]
Yes, being able to select data is key. My concern was on data quantity. Given that that is where the *.gl stack is aligned, I don't want a table component that chokes on large quantites of data. Pagination may help, but the issue as I see it is that you would need to get clever around maintaining selection between pages. So better to have a component that can simply handle a large number of rows.

**Comment by ngodber** [comment]
@morphoix I see you recently committed to Kepler. Would you be interested in being funded to implement this?
